lib/python/pathling/functions.py

```diff
@@ -0,0 +1,41 @@
+from pyspark.sql import Column
+from pyspark.sql.functions import lit, struct
+
+from pathling.coding import Coding
+
+
+def to_coding(coding_column: Column, system: str, version: str = None):
+    """
```

Comment:
Maybe it would be also useful to create codings where system is a column not literals.
This could either be done with a function like:

```def to_coding(coding: Union[Column, str], system:Union[Column, str] ... ) ```

and then testing of the actual type. The only problem is that this is somewhat inconsistent with spark where passing a string where a column is expected is interpreted as the name of the column.

So it could also be:

```def to_coding(coding_col: Column, system_col:Optional[Column] = None,  system_lit:Optional[str]  ) ```

and then call it like: 

```to_coding(col("code"), system_lit = 'http://loinc.org") ```

Reply:
I agree that this would be pretty easy to do, but I don't think that we have this requirement yet.

I think the scenario is that we have non-FHIR data with a system-specific column that contains a FHIR URI for the system (which wouldn't be the case in the case of say, OMOP, they use different URIs that would need to be translated anyway).

I think we should implement this feature on-demand.